firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileDownloadService.java

```diff
@@ -286,12 +332,17 @@ public File loadNewlyDownloadedModelFile(CustomModel model) throws FirebaseMlExc
     String downloadingModelHash = model.getModelHash();
 
     if (downloadingId == 0 || downloadingModelHash.isEmpty()) {
-      // no downloading model file or incomplete info.
+      // Clear the downloading info completely.
+      // It handles the case: developer clear the app cache but downloaded model file in
+      // DownloadManager's cache would not be cleared.
+      removeOrCancelDownload(model.getName(), model.getDownloadId());
       return null;
     }
 
     Integer statusCode = getDownloadingModelStatusCode(downloadingId);
     if (statusCode == null) {
+      // No status code, it may mean no such download or no download manager.
```

Comment:
Maybe you want to log this?

Reply:
Done. I'll be adding a bunch of logging probably early next week.